Source/CarthageKit/BuildArguments.swift

```diff
@@ -84,6 +84,11 @@ public struct BuildArguments {
 			args += [ "BITCODE_GENERATION_MODE=\(bitcodeGenerationMode.rawValue)" ]
 		}
 
+		if let carthagePath = NSBundle.mainBundle().executablePath
+			where NSBundle.mainBundle().executableURL?.lastPathComponent?.lowercaseString.rangeOfString("carthage") != nil {
```

Comment:
I truly don't remember my thinking in changing to use `executablePath` as well as `executableURL`, that should be fixed.

The use of `lowercaseString` is directly a result of using `rangeOfString()` and the reason for that was an explicit desire to catch the use of any binary with `carthage` in the name. I was envisioning the binary names being `carthage_0.14` or along those lines. Matching against a tool named `thisIsNotCarthage` would be an undesirable effect of course. I could conceivably use a proper regex to make sure `carthage` only featured at the beginning of the string but then someone might name the binary `carthageIsNotThisTool`. 

I'm not really sure if it is preferable to only catch the canonical name `carthage` or to allow a bit of leeway in case users rename their binaries. If it doesn't end up only working for `carthage` I will update it to use `localizedStandardContainsString()` rather than `lowerCaseString.rangeOfString()`.